packages/grid/x-data-grid/src/hooks/features/export/useGridPrintExport.tsx

```diff
@@ -53,27 +55,20 @@ export const useGridPrintExport = (
           return;
         }
 
-        const columnVisibilityModel = gridColumnVisibilityModelSelector(apiRef);
-        const columns = gridColumnDefinitionsSelector(apiRef);
+        const exportedColumnFields = getColumnsToExport({
```

Comment:
@alexfauquette I ran into an issue here.

I set `disableExport: true` on a column, but it still showed up in the print document.
I discovered that `getColumnsToExport` was never called because I didn't specify `fields` or `allColumns`, so this function returned a few rows above (`if (!fields && !allColumns) { ... return;`)

It seems to me like `disableExport` should still be checked even if `fields` and `allColumns` are not used.